app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt

```diff
@@ -65,10 +69,14 @@ class CreditCardEditorFragment : SecureFragment(R.layout.fragment_credit_card_ed
 
         val binding = FragmentCreditCardEditorBinding.bind(view)
 
-        creditCardEditorState =
-            args.creditCard?.toCreditCardEditorState(storage) ?: getInitialCreditCardEditorState()
-        creditCardEditorView = CreditCardEditorView(binding, interactor)
-        creditCardEditorView.bind(creditCardEditorState)
+        CoroutineScope(Dispatchers.Main).launch {
```

Comment:
Sorry for the dumb question, I am wondering what the difference between `CoroutineScope` vs `lifecycleScope.launch(Dispatchers.Main)`. How do we decide which one to use in these contexts?

Reply:
Yeah, good call - we do want `lifecycleScope` here. It's already bound to `Dispatchers.Main.immediate`, so just `lifecycleScope.launch { ... }` should be fine.

Since we have a fragment here and the work we may schedule is relevant in context of that fragment being around, tying it to the lifecycle of the fragment makes sense.